doc/doxygen/index.doc

```diff
@@ -1,265 +1,20 @@
-/*! \mainpage DRT/LinAlg: The discretization handling module and linear algebra subsystem
+/*! \mainpage 4C: A Comprehensive Multi-Physics Simulation Framework
 \htmlonly
 \endhtmlonly
 
-\section drt_intro Introduction
+\section Overview
 
-DRT provides fundamental services to construct and manage 2D/3D discretizations
-in parallel and serial. LinAlg provides parallel sparse and dense matrix/vector operations.
+4C (Comprehensive Computational Community Code) is a powerful research code for multi-physics computer simulations.
+Building on more than 20 years of research and development, 4C can model a plethora of physical systems, including solid mechanics, fluid dynamics, scalar transport, and chemical reaction among others.
 
-\section Overviewdrt Overview of DRT
+Large parts of 4C are based on finite element methods (FEM), but alternative discretization methods such as discontinuous Galerkin methods (DG), particle methods and mesh-free methods have also been successfully integrated. The research software is implemented throughout in object-oriented programming (C++) using modern software design and is parallelized with MPI for distributed memory hardware architectures.
```

Comment:
```suggestion
Large parts of 4C are based on finite element methods (FEM), but alternative discretization methods such as discontinuous Galerkin methods (DG), particle methods and mesh-free methods have also been successfully integrated. The research software is implemented throughout in C++ using modern software design and is parallelized with MPI for distributed memory hardware architectures.
```
We are using a lot object-oriented programming but that's in many cases not the best design decision.

Reply:
@amgebauer Sure, makes sense! But do we really use "modern software design"? Maybe not... what do you think?

Reply:
True, for me it's just a (often overused) phrase. What is modern now is typically outdated in a few years. I think we have things that one can call "modern", but we also have a lot of things that has been called "modern" more than a decade ago. Still, I don't see a direct need to actively remove it. It's just the doxygen start page, and we, for sure, aim at modern software design.